src/main.cpp

```diff
@@ -59,7 +59,7 @@ static std::string parse_commandline(int argc, char *argv[]) {
     v_desc.add_options()
         ("help,h", "Show commandline options.")
         ("threads,t", po::value<int>()->default_value
-                      (std::min(2, cfg_num_threads)),
+                      (std::min(cfg_num_threads, cfg_max_threads)),
```

Comment:
Should be simply (cfg_num_threads) I think?

Reply:
as discussed on discord cfg_max_threads can be 1, and I am not changing that behavior, but enforcing it, since it is called max_threads afterall. If we decide max threads can't be less than 2, we can fix that both places and simplify this.